curriculum/challenges/english/25-front-end-development/quiz-dynamic-programming/66edd5ffe490bef99ea99395.md

```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
+What is the main characteristic of problems that can be solved using greedy algorithms?
 
 #### --distractors--
 
-Placeholder distractor 1
+Problems that require backtracking to find the optimal solution
 
 ---
 
-Placeholder distractor 2
+Problems that have overlapping subproblems and require memoization
 
 ---
 
-Placeholder distractor 3
+Problems that can only be solved using brute force methods
 
 #### --answer--
 
-Placeholder answer
+They exhibit the greedy-choice property, meaning a global optimum can be arrived at by selecting the local optimum at each step
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the activity selection problem?
 
 #### --distractors--
 
-Placeholder distractor 1
+The activity selection problem involves finding the shortest path between two points
 
 ---
 
-Placeholder distractor 2
+The activity selection problem is used to sort a list of activities by their start times
 
 ---
 
-Placeholder distractor 3
+The activity selection problem can only be solved using dynamic programming
 
 #### --answer--
 
-Placeholder answer
+The activity selection problem involves selecting the maximum number of activities that don’t overlap, given their start and end times
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How does the greedy algorithm solve the activity selection problem?
 
 #### --distractors--
 
-Placeholder distractor 1
+The greedy algorithm selects activities based on their start times
 
 ---
 
-Placeholder distractor 2
+The greedy algorithm uses dynamic programming to find the optimal solution
 
 ---
 
-Placeholder distractor 3
+The greedy algorithm selects the longest duration activities first
 
 #### --answer--
 
-Placeholder answer
+By always selecting the next activity that finishes the earliest
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the time complexity of the greedy algorithm for the activity selection problem?
 
 #### --distractors--
 
-Placeholder distractor 1
+O(n^2)
 
 ---
 
-Placeholder distractor 2
+O(n^3)
 
 ---
 
-Placeholder distractor 3
+O(2^n)
 
 #### --answer--
 
-Placeholder answer
+O(n log n), due to the need to sort the activities by their finish times
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the difference between fractional and 0/1 knapsack problems?
 
 #### --distractors--
 
-Placeholder distractor 1
+Fractional knapsack problems can only be solved using dynamic programming.
 
 ---
 
-Placeholder distractor 2
+0/1 knapsack problems allow taking fractions of items
 
 ---
 
-Placeholder distractor 3
+Fractional knapsack problems are always solved using brute force methods
 
 #### --answer--
 
-Placeholder answer
+In the fractional knapsack problem, you can take fractions of items, while in the 0/1 knapsack problem, you must take whole items or none at all
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How can you compute the nth Fibonacci number using dynamic programming?
```

Comment:
Possible options from this question further are a bit repetitive. Answer is always the longest. Distractors are using similar wording, with changed details. Maybe some variety could be added here?

Reply:
ok